src/core/core.layouts.js

```diff
@@ -208,201 +327,75 @@ module.exports = {
-		// changed due to the application of the margins in step 6. Since we can only get bigger, this is safe to do
-		// without calling `fit` again
-		var newMaxChartAreaHeight = height - outerBoxSizes.top - outerBoxSizes.bottom;
-		var newMaxChartAreaWidth = width - outerBoxSizes.left - outerBoxSizes.right;
+		// First fit vertical layouts, starting from padding and no margins
+		outerDims = fitBoxes(verticalBoxes, chartArea, padding);
 
-		if (newMaxChartAreaWidth !== maxChartAreaWidth || newMaxChartAreaHeight !== maxChartAreaHeight) {
-			helpers.each(verticalBoxes, function(box) {
-				box.height = newMaxChartAreaHeight;
-			});
+		// Adjust chart area based on vertical layouts.
+		updateChartArea(chartArea, outerDims, maxPadding);
 
-			helpers.each(horizontalBoxes, function(box) {
-				if (!box.fullWidth) {
-					box.width = newMaxChartAreaWidth;
-				}
-			});
+		// Fit horizontal boxes, providing vertical box widths as margins
+		outerDims = fitBoxes(horizontalBoxes, chartArea, outerDims);
 
-			maxChartAreaHeight = newMaxChartAreaHeight;
-			maxChartAreaWidth = newMaxChartAreaWidth;
+		// Adjust chart area based on horizontal layouts
+		if (updateChartArea(chartArea, outerDims, maxPadding)) {
+			// if the area changed, re-fit vertical boxes
+			outerDims.left = padding.left;
+			outerDims.right = padding.right;
+			outerDims = fitBoxes(verticalBoxes, chartArea, outerDims);
+			updateChartArea(chartArea, outerDims, maxPadding);
 		}
 
-		// Step 7 - Position the boxes
-		var left = leftPadding + leftPaddingAddition;
-		var top = topPadding + topPaddingAddition;
-
-		function placeBox(box) {
-			if (box.isHorizontal()) {
-				box.left = box.fullWidth ? leftPadding : outerBoxSizes.left;
-				box.right = box.fullWidth ? width - rightPadding : outerBoxSizes.left + maxChartAreaWidth;
-				box.top = top;
-				box.bottom = top + box.height;
-
-				// Move to next point
-				top = box.bottom;
-
-			} else {
-
-				box.left = left;
-				box.right = left + box.width;
-				box.top = outerBoxSizes.top;
-				box.bottom = outerBoxSizes.top + maxChartAreaHeight;
-
-				// Move to next point
-				left = box.right;
+		helpers.each(horizontalBoxes, function(layout) {
+			if (!layout.box.fullWidth) {
+				layout.box.width = chartArea.w;
 			}
-		}
+		});
 
-		helpers.each(leftBoxes.concat(topBoxes), placeBox);
+		// Adjust top/left of outerDims and boxCorner with padding if needed
+		adjustWithPadding(outerDims, boxCorner, maxPadding);
 
-		// Account for chart width and height
-		left += maxChartAreaWidth;
-		top += maxChartAreaHeight;
+		// Finally place the layouts to correct coordinates
+		// - left / top
+		boxCorner = placeBoxes(boxes.lt, chartArea, outerDims, boxCorner);
 
-		helpers.each(rightBoxes, placeBox);
-		helpers.each(bottomBoxes, placeBox);
+		// Move to oppisite side of chart
+		boxCorner.left += chartArea.w;
+		boxCorner.top += chartArea.h;
+
+		// - right / bottom
+		placeBoxes(boxes.rb, chartArea, outerDims, boxCorner);
 
 		// Step 8
```

Comment:
"Step 8" doesn't mean anything since the step descriptions were deleted

Reply:
removed